pkg/controller/dgraph/models/container.go

```diff
@@ -66,16 +67,25 @@ func StoreAndRetrieveContainers(pod api_v1.Pod, podUID string) []*Container {
 
 		var container *Container
 		if containerUID == "" {
-			assigned, err := newContainer(containerXid, c.Name)
+			assigned, err := newContainer(containerXid, c.Name, pod.GetCreationTimestamp().Time)
 			containerUID = assigned.Uids["blank-0"]
 			if err != nil {
 				log.Errorf("Unable to create container: %s", containerXid)
 				continue
 			}
 		}
-		container = &Container{
-			ID:  dgraph.ID{UID: containerUID},
-			Pod: Pod{ID: dgraph.ID{UID: podUID}},
+
+		if isDeleted {
```

Comment:
use idDeleted flag just to update the endtime. We need to persist start time for container.